packages/@aws-cdk-testing/cli-integ/lib/shell.ts

```diff
@@ -27,37 +29,54 @@ export async function shell(command: string[], options: ShellOptions = {}): Prom
 
   const env = options.env ?? (options.modEnv ? { ...process.env, ...options.modEnv } : process.env);
 
-  const child = child_process.spawn(command[0], command.slice(1), {
+  // copy because we will be shifting it
+  const interact = [...(options.interact ?? [])];
+
+  const child = Process.spawn(command[0], command.slice(1), {
     ...options,
     env,
-    // Need this for Windows where we want .cmd and .bat to be found as well.
-    shell: true,
-    stdio: ['ignore', 'pipe', 'pipe'],
-  });
+    tty: interact.length > 0,
+  })
 
   return new Promise<string>((resolve, reject) => {
     const stdout = new Array<Buffer>();
     const stderr = new Array<Buffer>();
 
-    child.stdout!.on('data', chunk => {
+    const stdoutSinceLastPrompt = new Array<Buffer>();
+
+    child.onStdout(chunk => {
       if (show === 'always') {
-        writeToOutputs(chunk);
+        writeToOutputs(chunk.toString('utf-8'));
       }
       stdout.push(chunk);
+      stdoutSinceLastPrompt.push(chunk);
+
+      const interaction = interact[0];
+      if (interaction) {
+
+        if (interaction.prompt.test(Buffer.concat(stdoutSinceLastPrompt).toString('utf-8'))) {
```

Comment:
Not matching on a per-line basis anymore? This will get progressively slower over time...

Reply:
I wonder if can add a small delay before starting `cdk` and just write the entire input to `stdin` line by line. It would mean we won't match on output at all, the subprocess will just read the user inputs line by line when needed.

Reply:
I was thinking something like:

```ts
class Process {
  private lines = new Splitter('\n');

  private onOutput(chunk: Buffer) {
    this.lines.append(chunk);
     
    let line;
    while ((line = this.lines.consumeNext()) !== undefined) {
      if (interaction.matches(line)) {
        // ...
      }
    }
  }
}

class Splitter {
  constructor(private readonly separator: string) { }

  public append(chunk: Buffer) { }  
  public consumeNext(): string | undefined { }
}
```

Reply:
Addressed